### ISSUE TITLE
Do not preload response content

### ISSUE THREAD

**Issue by SP-l33t** [lexiforest/curl_cffi#465]
**The question**
I would like to not load the response body, unless I get resp code == 200.
Explaination: I'm limited proxies, and I would like to know if I can only load reponse body if needed. Sometimes I face 403 (CloudFlare) and the response is quite big, which might lead to additional expenses.
My only idea so far is to use `stream=True`, but I want to make sure if it's gonna work the way I described above.

**Documentation suggesion**

That's a separate one, but I couldn't find wether I can use socks4/5 proxies or not in the docs.

Thanks in advance!

**Comment by lexiforest** [comment]
I don't think so, that is not how a proxy server works. Anyway, the only way to know is to experiment by yourself. But I highly doubt it.

> I couldn't find wether I can use socks4/5 proxies or not in the docs.

The way you use proxies in requests, nothing special here.

**Comment by dolfies** [comment]
Can you use OPTIONS requests?
